benchmark/CMakeLists.txt

```diff
@@ -18,11 +18,19 @@ include(benchmark/dependencies/metal)
 include(benchmark/dependencies/mpl)
 include(benchmark/dependencies/std)
 
-macro(add_dataset datasets category case lib datatype path_to_template range)
+macro(add_dataset datasets category case lib datatype path_to_template k range)
+    set(env "{k: ${k}")
```

Comment:
> I'm not a big fan of specifying additional ENV values as a CMake list

Me neither, but I am even less a fan of the obvious alternative to be redundant and specify both `k` and `scale` and *regexping* a string to match key value pairs between curly braces is simply insane. 

It might not be pretty but that is the CMake syntax for pairs.